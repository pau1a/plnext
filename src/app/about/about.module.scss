.page {
  display: flex;
  flex-direction: column;
  gap: clamp(var(--space-2xl), 6vw, calc(var(--space-3xl) * 1.25));
  padding-block: clamp(var(--space-2xl), 8vw, calc(var(--space-3xl) * 1.5));
}

.hero {
  position: relative;
  overflow: hidden;
  border-radius: var(--radius-lg);
  padding: clamp(var(--space-xl), 8vw, calc(var(--space-3xl) * 1.1));
  background: linear-gradient(135deg, color-mix(in srgb, var(--color-teal-700) 38%, transparent) 0%,
      transparent 65%), var(--surface-elevated);
  box-shadow: var(--shadow-md);
  border: 1px solid var(--surface-border);
}

.hero::after {
  content: "";
  position: absolute;
  inset: 0;
  background: radial-gradient(circle at top left,
      color-mix(in srgb, var(--color-teal-500) 18%, transparent) 0%,
      transparent 55%);
  pointer-events: none;
}

.heroInner {
  position: relative;
  display: flex;
  flex-direction: column;
  gap: var(--space-lg);
  max-width: min(640px, 100%);
  color: var(--text-primary);
}

.eyebrow {
  font-size: var(--font-size-xs);
  font-weight: 600;
  letter-spacing: 0.32em;
  text-transform: uppercase;
  color: color-mix(in srgb, var(--color-teal-600) 75%, var(--text-primary));
}

.title {
  font-size: clamp(var(--font-size-3xl), 6vw, var(--font-size-4xl));
  font-weight: 700;
  color: var(--color-teal-700);
  margin: 0;
}

:global(body.dark) .title {
  color: var(--color-teal-200);
}

.subhead {
  font-size: var(--font-size-lg);
  color: var(--text-muted);
  margin: 0;
  line-height: var(--line-height-snug);
}

.layout {
  display: grid;
  gap: clamp(var(--space-xl), 4vw, var(--space-3xl));
}

@media (min-width: 960px) {
  .layout {
    grid-template-columns: minmax(280px, 360px) minmax(0, 1fr);
    align-items: start;
  }
}

.portraitCard {
  position: relative;
  border-radius: var(--radius-lg);
  background: var(--surface-elevated);
  border: 1px solid var(--surface-border);
  box-shadow: var(--shadow-sm);
  padding: var(--space-lg);
  display: flex;
  flex-direction: column;
  gap: var(--space-lg);
  transition: transform var(--transition-base), box-shadow var(--transition-base);
}

.portraitCard:hover,
.portraitCard:focus-within {
  transform: translateY(-4px);
  box-shadow: var(--shadow-md);
}

.portraitImageWrapper {
  position: relative;
  border-radius: calc(var(--radius-lg) - var(--space-xs));
  overflow: hidden;
  aspect-ratio: 4 / 5;
  background: color-mix(in srgb, var(--color-graphite-200) 40%, transparent);
}

.portraitImage {
  width: 100%;
  height: 100%;
  object-fit: cover;
  display: block;
}

.badgeRow {
  padding-block: var(--space-xs);
}

.bio {
  display: flex;
  flex-direction: column;
  gap: var(--space-lg);
  color: var(--text-primary);
  font-size: var(--font-size-md);
  line-height: var(--line-height-base);
}

.bio h2 {
  font-size: clamp(var(--font-size-xl), 3vw, var(--font-size-2xl));
  color: var(--color-teal-700);
  margin: 0;
}

:global(body.dark) .bio h2 {
  color: var(--color-teal-200);
}

.bio p {
  margin: 0;
}

.constantsSection {
  display: flex;
  flex-direction: column;
  gap: var(--space-xl);
}

.constantsHeading {
  font-size: clamp(var(--font-size-xl), 3vw, var(--font-size-2xl));
  margin: 0;
  color: var(--color-teal-700);
}

:global(body.dark) .constantsHeading {
  color: var(--color-teal-200);
}

.constantsGrid {
  display: grid;
  gap: var(--space-lg);
}

@media (min-width: 768px) {
  .constantsGrid {
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }
}

.closingCard {
  border-radius: var(--radius-lg);
  background: var(--surface-elevated);
  border: 1px solid var(--accent-border);
  padding: clamp(var(--space-xl), 4vw, var(--space-2xl));
  box-shadow: var(--shadow-sm);
  position: relative;
  color: var(--text-primary);
  font-size: var(--font-size-lg);
  line-height: var(--line-height-snug);
}

.closingCard::before {
  content: "";
  position: absolute;
  inset-inline-start: 0;
  inset-block: var(--space-md);
  width: 4px;
  border-radius: 999px;
  background: linear-gradient(180deg, var(--color-crimson-400), var(--color-crimson-600));
}

.closingBody {
  margin: 0;
  padding-inline-start: var(--space-lg);
}

@media (min-width: 768px) {
  .closingBody {
    padding-inline-start: calc(var(--space-lg) + 2px);
  }
}
