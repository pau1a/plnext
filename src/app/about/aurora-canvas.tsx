"use client";

import clsx from "clsx";
import { useEffect, useRef } from "react";

import { usePrefersReducedMotion } from "@/lib/motion";

type AuroraCanvasProps = {
  containerRef: React.RefObject<HTMLElement>;
  className?: string;
};

const MAX_DPR = 2;

export default function AuroraCanvas({ containerRef, className }: AuroraCanvasProps) {
  const canvasRef = useRef<HTMLCanvasElement | null>(null);
  const frameRef = useRef<number>();
  const pointerRef = useRef({ x: 0.5, y: 0.35 });
  const pointerTargetRef = useRef({ x: 0.5, y: 0.35 });
  const scrollRef = useRef(0);
  const timeRef = useRef(0);

  const prefersReducedMotion = usePrefersReducedMotion();

  useEffect(() => {
    const canvas = canvasRef.current;
    const host = containerRef.current;
    if (!canvas || !host) {
      return;
    }

    const context = canvas.getContext("2d");
    if (!context) {
      return;
    }

    let width = 0;
    let height = 0;
    const dpr = Math.min(window.devicePixelRatio || 1, MAX_DPR);

    const resize = () => {
      const container = containerRef.current;
      if (!canvas || !container) {
        return;
      }

      const rect = container.getBoundingClientRect();
      width = Math.max(1, Math.round(rect.width));
      height = Math.max(1, Math.round(rect.height));

      canvas.width = width * dpr;
      canvas.height = height * dpr;
      canvas.style.width = `${width}px`;
      canvas.style.height = `${height}px`;

      const maybeReset = (context as CanvasRenderingContext2D & {
        resetTransform?: () => void;
      }).resetTransform;

      if (typeof maybeReset === "function") {
        maybeReset.call(context);
      } else {
        context.setTransform(1, 0, 0, 1, 0, 0);
      }

      context.scale(dpr, dpr);
    };

    resize();
    window.addEventListener("resize", resize);

    const updatePointer = (event: PointerEvent) => {
      const container = containerRef.current;
      if (!container) {
        return;
      }
      const rect = container.getBoundingClientRect();
      if (rect.width === 0 || rect.height === 0) {
        return;
      }
      pointerTargetRef.current = {
        x: (event.clientX - rect.left) / rect.width,
        y: (event.clientY - rect.top) / rect.height,
      };
    };

    const resetPointer = () => {
      pointerTargetRef.current = { x: 0.5, y: 0.35 };
    };

    window.addEventListener("pointermove", updatePointer, { passive: true });
    window.addEventListener("pointerleave", resetPointer);

    const renderStatic = () => {
      if (width === 0 || height === 0) {
        return;
      }

      context.clearRect(0, 0, width, height);

      const base = context.createLinearGradient(0, 0, 0, height);
      base.addColorStop(0, "#083947");
      base.addColorStop(0.45, "#105a60");
      base.addColorStop(1, "#0a1f2c");
      context.fillStyle = base;
      context.fillRect(0, 0, width, height);

      const overlay = context.createRadialGradient(width * 0.6, height * 0.35, 0, width * 0.6, height * 0.35, height * 0.65);
      overlay.addColorStop(0, "rgba(87, 227, 197, 0.55)");
      overlay.addColorStop(1, "rgba(87, 227, 197, 0)");
      context.fillStyle = overlay;
      context.fillRect(0, 0, width, height);
    };

    if (prefersReducedMotion) {
      renderStatic();
      return () => {
        window.removeEventListener("resize", resize);
        window.removeEventListener("pointermove", updatePointer);
        window.removeEventListener("pointerleave", resetPointer);
      };
    }

    const draw = () => {
      const container = containerRef.current;
      if (!container) {
        frameRef.current = requestAnimationFrame(draw);
        return;
      }

      const rect = container.getBoundingClientRect();
      const viewportHeight = window.innerHeight || 1;
      const visible =
        rect.height + viewportHeight === 0
          ? 0
          : Math.min(
              1,
              Math.max((viewportHeight - rect.top) / (rect.height + viewportHeight), 0),
            );

      scrollRef.current += (visible - scrollRef.current) * 0.05;

      pointerRef.current.x += (pointerTargetRef.current.x - pointerRef.current.x) * 0.08;
      pointerRef.current.y += (pointerTargetRef.current.y - pointerRef.current.y) * 0.08;

      timeRef.current += 0.008;

      context.clearRect(0, 0, width, height);

      const base = context.createLinearGradient(0, 0, 0, height);
      base.addColorStop(0, "#041c2b");
      base.addColorStop(0.4, "#0b3d45");
      base.addColorStop(1, "#0b1a24");
      context.fillStyle = base;
      context.fillRect(0, 0, width, height);

      const pointerX = pointerRef.current.x;
      const pointerY = pointerRef.current.y;
      const scroll = scrollRef.current;

      const waves = [
        { hue: 168, offset: 0.15, intensity: 0.65, drift: 0.6 },
        { hue: 198, offset: 0.45, intensity: 0.55, drift: 0.9 },
        { hue: 350, offset: 0.78, intensity: 0.4, drift: 1.2 },
      ];

      for (let index = 0; index < waves.length; index += 1) {
        const wave = waves[index];
        const sine = Math.sin(timeRef.current * wave.drift + index);
        const cx =
          width *
          (wave.offset + (pointerX - 0.5) * 0.35 + sine * 0.08 + scroll * 0.12);
        const cy =
          height *
          (0.25 + pointerY * 0.25 + index * 0.18 + scroll * (0.08 + index * 0.04));
        const radius = height * (0.65 + index * 0.3 + scroll * 0.25);

        const gradient = context.createRadialGradient(cx, cy, 0, cx, cy, radius);
        const alpha = Math.min(0.72, 0.45 + scroll * 0.35);
        gradient.addColorStop(
          0,
          `hsla(${wave.hue + sine * 12}, 82%, 68%, ${alpha * wave.intensity})`,
        );
        gradient.addColorStop(
          1,
          `hsla(${wave.hue + 30}, 75%, 52%, 0)`,
        );

        context.globalCompositeOperation = "lighter";
        context.fillStyle = gradient;
        context.fillRect(0, 0, width, height);
        context.globalCompositeOperation = "source-over";
      }

      const mist = context.createLinearGradient(0, height * 0.55, 0, height);
      mist.addColorStop(0, "rgba(15, 66, 86, 0.18)");
      mist.addColorStop(1, "rgba(15, 66, 86, 0)");
      context.fillStyle = mist;
      context.fillRect(0, 0, width, height);

      frameRef.current = requestAnimationFrame(draw);
    };

    frameRef.current = requestAnimationFrame(draw);

    return () => {
      if (frameRef.current) {
        cancelAnimationFrame(frameRef.current);
      }
      window.removeEventListener("resize", resize);
      window.removeEventListener("pointermove", updatePointer);
      window.removeEventListener("pointerleave", resetPointer);
    };
  }, [containerRef, prefersReducedMotion]);

  return (
    <canvas
      ref={canvasRef}
      className={clsx("about-aurora", className)}
      aria-hidden="true"
    />
  );
}
