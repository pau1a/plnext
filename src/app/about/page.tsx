import type { Metadata } from "next";
import AboutPageContent from "./about-content";

const BADGES = [
  "Industrial automation",
  "AI security",
  "OT & networks",
  "Python-first",
  "Measurement-led",
] as const;

const BIOGRAPHY = [
  "My career began in military radio and satellite systems and matured through large-scale IP networking. Today I focus on AI-enabled operations for industrial environments. I design layered defences, instrument estates so risk is measured not guessed, and remove toil with code so teams can focus on the hard problems.",
  "I shape security by design for OT and automation programmes and align controls to recognised standards such as IEC 62443. I am the person teams tap to untangle cross-discipline problems when they cut across layers and refuse to fit into one box.",
  "Over the past year I have gone deep on Python-first machine learning and modern data tooling. I build guardrails that let models survive contact with production, including secure MLOps, model and data lineage, drift detection, and adversarial robustness.",
  "Postgraduate study sharpened the theory. I hold an MSc Cyber Security and a PGDip Cyber. My dissertation explored how cryptographic transparency and distributed ledgers can strengthen identity and provenance across industrial supply chains. Applied cryptography is part of the toolbox. I use signatures, attestations and transparency logs, with blockchain only where it reduces risk and friction for integrity, provenance and tamper-evident logging.",
  "My operating style is measurement-led. Results look like lower MTTR, fewer false positives, safer change windows and quieter on-call. I prefer simple, observable architectures to clever, brittle ones. Controls and infrastructure should be code. Runbooks should be testable. Recovery should be unremarkable.",
  "Earlier in my journey I founded an AI venture and chose to wind it down when the stack was not ready. That experience sharpened timing and judgement and reinforced a bias for pragmatic engineering.",
] as const;

const PRINCIPLES = [
  {
    title: "Narrow the blast radius",
    body: "Design for safe failure. Isolate. Contain. Make incidents small and boring.",
  },
  {
    title: "Increase visibility",
    body: "Instrument the estate. Measure risk rather than guess it. Logs that tell the truth.",
  },
  {
    title: "Shorten recovery",
    body: "Lower MTTR with clear runbooks, good alerts, and tested recovery paths.",
  },
  {
    title: "Keep people out of headlines",
    body: "Protect teams and users by making systems resilient and decisions auditable.",
  },
] as const;

const CLOSING_STATEMENT =
  "If the future is AI and automation, it needs engineers who can secure it end to end. That is my lane.";

const siteUrl = process.env.NEXT_PUBLIC_SITE_URL ?? "https://paulalivingstone.com";

const PERSON_JSON_LD = {
  "@context": "https://schema.org",
  "@type": "Person",
  name: "Paula Livingstone",
  jobTitle: "Engineer, Industrial Cybersecurity",
  image: "https://cdn.networklayer.co.uk/paulalivingstone/images/plprof.jpeg",
  url: `${siteUrl}/about`,
  inLanguage: "en-GB",
} as const;

export const metadata: Metadata = {
  title: "About | Paula Livingstone",
  description:
    "Industrial automation and AI security. Narrow the blast radius, increase visibility, shorten recovery.",
  alternates: {
    canonical: "/about",
  },
  openGraph: {
    title: "About | Paula Livingstone",
    description:
      "Industrial automation and AI security. Narrow the blast radius, increase visibility, shorten recovery.",
    url: `${siteUrl}/about`,
    type: "profile",
    images: [
      {
        url: `${siteUrl}/window.svg`,
        width: 1200,
        height: 630,
        alt: "Paula Livingstone window mark",
      },
    ],
  },
  twitter: {
    card: "summary_large_image",
    title: "About | Paula Livingstone",
    description:
      "Industrial automation and AI security. Narrow the blast radius, increase visibility, shorten recovery.",
    images: [`${siteUrl}/window.svg`],
  },
};

export default function AboutPage() {
  return (
    <>
      <script type="application/ld+json" dangerouslySetInnerHTML={{ __html: JSON.stringify(PERSON_JSON_LD) }} />
      <AboutPageContent
        badges={BADGES}
        biography={BIOGRAPHY}
        principles={PRINCIPLES}
        closingStatement={CLOSING_STATEMENT}
      />
    </>
  );
}
