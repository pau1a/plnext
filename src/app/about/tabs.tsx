"use client";

import Link from "next/link";
import { usePathname } from "next/navigation";

const items = [
  { href: "/about/about1", label: "About 1" },
  { href: "/about/about2", label: "About 2" },
  { href: "/about/about3", label: "About 3" },
  { href: "/about/about4", label: "About 4" },
  { href: "/about/about5", label: "About 5" },
];

export function Tabs() {
  const pathname = usePathname();

  return (
    <nav className="rounded-2xl p-2 bg-[hsl(var(--teal-50))] dark:bg-[hsl(var(--teal-900)/0.3)] ring-1 ring-black/5 dark:ring-white/10">
      <ul className="grid grid-cols-5 gap-2">
        {items.map((item) => {
          const active = pathname?.startsWith(item.href);

          return (
            <li key={item.href}>
              <Link
                href={item.href}
                className={[
                  "block rounded-xl px-3 py-2 text-sm md:text-[15px] focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-[hsl(var(--crimson-500))] transition",
                  active
                    ? "bg-[hsl(var(--teal-200))] text-[hsl(var(--teal-900))] dark:bg-[hsl(var(--teal-800))] dark:text-[hsl(var(--teal-100))] shadow"
                    : "bg-white/70 dark:bg-neutral-900/40 hover:bg-white/90 dark:hover:bg-neutral-900/60",
                ].join(" ")}
              >
                {item.label}
              </Link>
            </li>
          );
        })}
      </ul>
    </nav>
  );
}
