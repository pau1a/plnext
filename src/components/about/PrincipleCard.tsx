import React from "react";
import clsx from "clsx";
import type { HTMLAttributes, ReactNode } from "react";

import styles from "./principle-card.module.scss";

type PrincipleCardProps = {
  title: string;
  body: ReactNode;
} & HTMLAttributes<HTMLDivElement>;

export function PrincipleCard({ title, body, className, ...props }: PrincipleCardProps) {
  return (
    <div className={clsx(styles.card, className)} {...props}>
      <h3 className={styles.title}>{title}</h3>
      <p className={styles.body}>{body}</p>
    </div>
  );
}

export default PrincipleCard;
