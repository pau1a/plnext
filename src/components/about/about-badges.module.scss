.badgeList {
  display: flex;
  flex-wrap: wrap;
  gap: var(--space-sm);
  margin: 0;
  padding: 0;
  list-style: none;
}

.badgeItem {
  margin: 0;
}

.badge {
  display: inline-flex;
  align-items: center;
  padding: var(--space-xs) var(--space-md);
  border-radius: 999px;
  background: color-mix(in srgb, var(--color-teal-300) 70%, var(--surface-elevated));
  color: color-mix(in srgb, var(--color-graphite-900) 82%, var(--text-primary));
  font-size: var(--font-size-sm);
  font-weight: 600;
  line-height: 1.4;
  letter-spacing: 0.01em;
  box-shadow: inset 0 0 0 1px color-mix(in srgb, var(--color-teal-500) 28%, transparent);
  transition: transform var(--transition-base), box-shadow var(--transition-base);
}

.badge:hover {
  transform: translateY(-2px);
  box-shadow: 0 0 0 2px color-mix(in srgb, var(--color-crimson-400) 35%, transparent),
    var(--shadow-sm);
}

.badge:focus-visible {
  outline: none;
  box-shadow: 0 0 0 3px color-mix(in srgb, var(--color-crimson-400) 60%, transparent);
}

:global(body.dark) .badge {
  background: color-mix(in srgb, var(--color-teal-700) 65%, var(--surface-elevated));
  color: var(--text-on-inverse);
  box-shadow: inset 0 0 0 1px color-mix(in srgb, var(--color-teal-300) 30%, transparent);
}

:global(body.dark) .badge:hover {
  box-shadow: 0 0 0 2px color-mix(in srgb, var(--color-crimson-300) 45%, transparent),
    var(--shadow-sm);
}

:global(body.dark) .badge:focus-visible {
  box-shadow: 0 0 0 3px color-mix(in srgb, var(--color-crimson-300) 70%, transparent);
}
