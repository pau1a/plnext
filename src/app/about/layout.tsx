import type { ReactNode } from "react";

import { Tabs } from "./tabs";

export default function AboutLayout({ children }: { children: ReactNode }) {
  return (
    <div className="container mx-auto max-w-6xl px-4 py-8">
      <Tabs />
      <div className="mt-4">{children}</div>
    </div>
  );
}
