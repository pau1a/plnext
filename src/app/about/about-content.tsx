"use client";

import { useEffect, useMemo } from "react";
import { motion, useReducedMotion } from "framer-motion";

import AboutBadges from "@/components/about/AboutBadges";
import PrincipleCard from "@/components/about/PrincipleCard";
import {
  createFadeInUpVariants,
  motionDurations,
  motionEasings,
  useMotionVariants,
  viewportDefaults,
} from "@/lib/motion";
import analytics from "@/lib/analytics/posthog";

import styles from "./about.module.scss";

type Principle = {
  title: string;
  body: string;
};

type AboutPageContentProps = {
  badges: readonly string[];
  biography: readonly string[];
  principles: readonly Principle[];
  closingStatement: string;
};

const heroVariants = createFadeInUpVariants({
  duration: motionDurations.base,
  offset: 16,
});

const heroEyebrowVariants = createFadeInUpVariants({
  duration: motionDurations.long,
  offset: 24,
});

const heroTitleVariants = createFadeInUpVariants({
  delay: 0.04,
  duration: motionDurations.long,
  offset: 24,
});

const heroSubheadVariants = createFadeInUpVariants({
  delay: 0.08,
  duration: motionDurations.long,
  offset: 24,
});

const heroCopyVariants = createFadeInUpVariants({
  delay: 0.16,
  duration: motionDurations.long,
  offset: 24,
});

const columnVariants = createFadeInUpVariants({
  duration: motionDurations.long,
  offset: 32,
});

const constantsCardVariants = createFadeInUpVariants({
  duration: motionDurations.base,
  offset: 24,
});

export function AboutPageContent({ badges, biography, principles, closingStatement }: AboutPageContentProps) {
  const { variants: heroMotion, shouldReduceMotion } = useMotionVariants(heroVariants);
  const heroEyebrowMotion = useMotionVariants(heroEyebrowVariants).variants;
  const heroTitleMotion = useMotionVariants(heroTitleVariants).variants;
  const heroSubheadMotion = useMotionVariants(heroSubheadVariants).variants;
  const heroCopyMotion = useMotionVariants(heroCopyVariants).variants;
  const columnMotion = useMotionVariants(columnVariants).variants;
  const constantsMotion = useMotionVariants(constantsCardVariants).variants;

  const reducedMotion = useReducedMotion();

  useEffect(() => {
    const tracker = (analytics as unknown as { track?: (event: string, properties?: Record<string, unknown>) => void }).track;
    if (typeof tracker === "function") {
      tracker("about_viewed");
      return;
    }

    if (typeof analytics.capture === "function") {
      analytics.capture("about_viewed");
    }
  }, []);

  const gridMotion = useMemo(() => {
    if (shouldReduceMotion || reducedMotion) {
      return {
        hidden: { opacity: 1 },
        visible: { opacity: 1 },
      };
    }

    return {
      hidden: { opacity: 0, y: 16 },
      visible: {
        opacity: 1,
        y: 0,
        transition: {
          duration: motionDurations.base,
          ease: motionEasings.standard,
          staggerChildren: 0.18,
          delayChildren: 0.12,
        },
      },
    };
  }, [reducedMotion, shouldReduceMotion]);

  return (
    <div className="l-container">
      <div className={styles.page}>
        <motion.section
          className={styles.hero}
          variants={heroMotion}
          initial={shouldReduceMotion ? "visible" : "hidden"}
          animate="visible"
        >
          <div className={styles.heroInner}>
            <motion.span className={styles.eyebrow} variants={heroEyebrowMotion}>
              ABOUT
            </motion.span>
            <motion.h1 className={styles.title} variants={heroTitleMotion}>
              Building calm in complex systems
            </motion.h1>
            <motion.p className={styles.subhead} variants={heroSubheadMotion}>
              I am an engineer who helps build and secure automated systems and the interconnected networks that run them.
              My focus is where operational technology, networks, and AI-enabled automation overlap. My job is to narrow the
              blast radius.
            </motion.p>
          </div>
        </motion.section>

        <motion.section
          className={styles.layout}
          variants={gridMotion}
          initial={shouldReduceMotion || reducedMotion ? "visible" : "hidden"}
          animate="visible"
        >
          <motion.div className={styles.portraitCard} variants={columnMotion}>
            <div className={styles.portraitImageWrapper}>
              {/* eslint-disable-next-line @next/next/no-img-element */}
              <img
                src="https://cdn.networklayer.co.uk/paulalivingstone/images/plprof.jpeg"
                alt="Paula Livingstone"
                className={styles.portraitImage}
                loading="lazy"
                width={800}
                height={1000}
              />
            </div>
            <div className={styles.badgeRow}>
              <AboutBadges labels={badges} />
            </div>
          </motion.div>

          <motion.article className={styles.bio} variants={columnMotion} aria-labelledby="about-story-heading">
            <h2 id="about-story-heading">From RF to AI-secured automation</h2>
            {biography.map((paragraph) => (
              <p key={paragraph}>{paragraph}</p>
            ))}
          </motion.article>
        </motion.section>

        <section className={styles.constantsSection} aria-labelledby="constants-heading">
          <motion.h2
            className={styles.constantsHeading}
            variants={heroCopyMotion}
            initial="hidden"
            whileInView="visible"
            viewport={viewportDefaults}
          >
            Constants
          </motion.h2>
          <div className={styles.constantsGrid}>
            {principles.map((principle) => (
              <motion.div
                key={principle.title}
                variants={constantsMotion}
                initial="hidden"
                whileInView="visible"
                viewport={viewportDefaults}
              >
                <PrincipleCard title={principle.title} body={principle.body} />
              </motion.div>
            ))}
          </div>
        </section>

        <motion.section
          className={styles.closingCard}
          variants={heroCopyMotion}
          initial="hidden"
          whileInView="visible"
          viewport={viewportDefaults}
        >
          <p className={styles.closingBody}>{closingStatement}</p>
        </motion.section>
      </div>
    </div>
  );
}

export default AboutPageContent;
