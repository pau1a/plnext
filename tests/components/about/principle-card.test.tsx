import React from "react";
import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import { axe } from "vitest-axe";

import { PrincipleCard } from "@/components/about/PrincipleCard";

describe("PrincipleCard", () => {
  const title = "Narrow the blast radius";
  const body = "Design for safe failure. Isolate. Contain. Make incidents small and boring.";

  it("renders the provided title and body", () => {
    render(<PrincipleCard title={title} body={body} />);

    expect(screen.getByRole("heading", { name: title })).toBeInTheDocument();
    expect(screen.getByText(body)).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { container } = render(<PrincipleCard title={title} body={body} />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it("passes basic accessibility checks", async () => {
    const { container } = render(<PrincipleCard title={title} body={body} />);

    const results = await axe(container);

    expect(results.violations).toHaveLength(0);
  });
});
