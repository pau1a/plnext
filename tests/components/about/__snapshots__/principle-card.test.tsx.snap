// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PrincipleCard > matches the snapshot 1`] = `
<div
  class="_card_26746a"
>
  <h3
    class="_title_26746a"
  >
    Narrow the blast radius
  </h3>
  <p
    class="_body_26746a"
  >
    Design for safe failure. Isolate. Contain. Make incidents small and boring.
  </p>
</div>
`;
